File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float n;

    do
    {
        n = get_float("Change owed: ");
    }
    while (n < 0);

    int dollars = round(n * 100);
    int quarters = 25;
    int dimes = 10;
    int nickels = 5;
    int pennies = 1;
    int qcount = 0;
    int dcount = 0;
    int ncount = 0;
    int pcount = 0;
    int coins = 0;

    while (dollars >= quarters)
    {
        dollars = dollars - 25; qcount++; coins++;
    }

    while (dollars >= dimes)
    {
        dollars = dollars - 10; dcount++; coins++;
    }

    while (dollars >= nickels)
    {
        dollars = dollars - 5; ncount++; coins++;
    }

    while (dollars >= pennies)
    {
        dollars = dollars - 1; pcount++; coins++;
    }

    printf("Total coins used: %i\n", coins);
    printf("Type of coins used:");
    printf(" %i quarters,", qcount);
    printf(" %i dimes,", dcount);
    printf(" %i nickels,", ncount);
    printf(" %i pennies.\n", pcount);
}

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    n = get_float("Change Due: ")
    if n > 0:
        break

dollars = round(n * 100)
quarters = 25
dimes = 10
nickels = 5
pennies = 1

coins = 0
qcount = 0
dcount = 0
ncount = 0
pcount = 0

while (dollars >= quarters):
    dollars = dollars - 25
    qcount += 1
    coins += 1

while (dollars >= dimes):
    dollars = dollars - 10
    dcount += 1
    coins += 1

while (dollars >= nickels):
    dollars = dollars - 5
    ncount += 1
    coins += 1

while (dollars >= pennies):
    dollars = dollars - 1
    pcount += 1
    coins += 1

print("Total coins used: ", end="")
print(coins)
print()
print("Type of coins used:")
print("Quarters: ", end="")
print(qcount)
print("Dimes: ", end="")
print(dcount)
print("Nickels: ", end="")
print(ncount)
print("Pennies: ", end="")
print(pcount)
print()